js/test/BTCUtils.test.js

```diff
@@ -1,442 +1,317 @@
+      const arraysAreEqual = utils.typedArraysAreEqual(res, extractMerkleRootBE[i].output);
+      assert.isTrue(arraysAreEqual);
+    }
   });
 
   it('extracts the target from a header', () => {
-    const res = BTCUtils.extractTarget(HEADER_170);
-    assert.equal(res, BigInt('26959535291011309493156476344723991336010898738574164086137773096960'));
+    const res = BTCUtils.extractTarget(extractTarget[0].input);
+    assert.equal(res, BigInt(26959535291011309493156476344723991336010898738574164086137773096960));
+    // this isn't working
+    // assert.equal(res, utils.bytesToUint(extractTarget[0].output));
   });
 
   it('extracts the prev block hash', () => {
-    const res = BTCUtils.extractPrevBlockBE(HEADER_170);
-    const u8aValue = HEADER.PREV_BLOCK_HASH;
-    const arraysAreEqual = utils.typedArraysAreEqual(res, u8aValue);
-    assert.isTrue(arraysAreEqual);
+    for (let i = 0; i < BTCUtils.extractPrevBlockBE.length; i += 1) {
+      const res = BTCUtils.extractPrevBlockBE(extractPrevBlockBE[i].input);
+      const arraysAreEqual = utils.typedArraysAreEqual(res, extractPrevBlockBE[i].output);
+      assert.isTrue(arraysAreEqual);
+    }
   });
 
   it('extracts a timestamp from a header', () => {
-    const res = BTCUtils.extractTimestamp(HEADER_170);
-    assert.equal(res, BigInt(HEADER.TIMESTAMP));
+    for (let i = 0; i < extractTimestamp.length; i += 1) {
+      const res = BTCUtils.extractTimestamp(extractTimestamp[i].input);
+      assert.equal(res, BigInt(extractTimestamp[i].output));
+    }
   });
 
   it('verifies a bitcoin merkle root', () => {
-    let res;
-    res = BTCUtils.verifyHash256Merkle(MERKLE_ROOT.TRUE[0], 0); // 0-indexed
-    assert.isTrue(res);
-
-    res = BTCUtils.verifyHash256Merkle(MERKLE_ROOT.TRUE[1], 1); // 0-indexed
-    assert.isTrue(res);
-
-    res = BTCUtils.verifyHash256Merkle(MERKLE_ROOT.TRUE[2], 4); // 0-indexed
-    assert.isTrue(res);
-
-    res = BTCUtils.verifyHash256Merkle(OP_RETURN_PROOF, OP_RETURN_INDEX);
-    assert.isTrue(res);
-
-    res = BTCUtils.verifyHash256Merkle(TWO_IN.PROOF, Number(TWO_IN.INDEX));
-    assert.isTrue(res);
-
-    // not evenly divisible by 32
-    res = BTCUtils.verifyHash256Merkle(MERKLE_ROOT.FALSE[0], 0);
-    assert.isFalse(res);
-
-    // 1-hash special case
-    res = BTCUtils.verifyHash256Merkle(MERKLE_ROOT.FALSE[1], 0);
-    assert.isTrue(res);
-
-    // 2-hash special case
-    res = BTCUtils.verifyHash256Merkle(MERKLE_ROOT.FALSE[2], 0);
-    assert.isFalse(res);
+    for (let i = 0; i < verifyHash256Merkle.length; i += 1) {
+      const res = BTCUtils.verifyHash256Merkle(
+        verifyHash256Merkle[i].input.proof,
+        verifyHash256Merkle[i].input.index // 0-indexed
+      );
+      if (verifyHash256Merkle[i].output) {
+        assert.isTrue(res);
+      } else {
+        assert.isFalse(res);
+      }
+    }
   });
 
   it('determines VarInt data lengths correctly', () => {
     let res;
 
+    // How can I store 0x01 in json?
```

Comment:
these are numbers, so you can do `1` for 0x01, `255` for 0xff, `254` for 0xfe, `253` for `0xfd`